src/Compilers/Core/MSBuildTask/Microsoft.Build.Tasks.CodeAnalysis.csproj

```diff
@@ -75,12 +72,12 @@
     </EmbeddedResource>
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.Build" Version="$(MicrosoftBuildFixedVersion)" />
-    <PackageReference Include="Microsoft.Build.Tasks.Core" Version="$(MicrosoftBuildTasksCoreFixedVersion)" />
+    <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildFrameworkVersion)" ExcludeAssets="Runtime" />
```

Comment:
I didn't understand what `ExcludeAssets` is for. What purpose does it serve here?

Reply:
A NuGet package can come with a variety of assets inside it. For this line there are two types that are important: runtime and compile time. The compile time assets are the libraries that we compile against (ref dlls for example). The runtime assets are items the NuPkg believes are necessary at runtime.

For MSBuild we are hosted inside their process. We never start up MSBuild on our own and create say a workspace. This bulid task is always loaded in their runtime. Hence their runtime assets aren't useful to us (and can possibly confuse our analysis). Hence we exclude them. 

Note: this is how we handle MSBuild assets in other parts of the Roslyn repository. The build task was always "wrong-ish" here.

Reply:
Should we also use `ExcludeAssets` below (in `Microsoft.Build.Tasks.CodeAnalysis.UnitTests.csproj`)?

Reply:
No. The unit tests need the msbuild dll runtime assets because they load the task outside of an msbuild process.